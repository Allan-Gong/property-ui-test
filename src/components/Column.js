import React from 'react';
import PropTypes from "prop-types";
import { Grid, Segment } from "semantic-ui-react";

const Column = ({ properties }) => (
  <Grid.Column>
    <Segment>{properties}</Segment>
  </Grid.Column>
);

Column.propTypes = {
  properties: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Column;
