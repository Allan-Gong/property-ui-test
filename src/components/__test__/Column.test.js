import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Column from '../Column';

describe('Column', () => {
  it('Column renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Column properties={[]} />, div);
  });

  it('should render correctly with given props', () => {
    const properties = [(<div key="1">test</div>)];
    
    const output = shallow(
      <Column properties={properties} />
    );

    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
