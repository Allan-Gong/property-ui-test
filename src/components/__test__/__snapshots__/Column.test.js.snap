// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render correctly with given props 1`] = `
<GridColumn>
  <Segment>
    <div
      key="1"
    >
      test
    </div>
  </Segment>
</GridColumn>
`;
